doc/requirements.txt

```diff
@@ -0,0 +1,61 @@
+.. _requirements:
+
+
+Requirements
+============
+
+Theano is mainly develeopped on 64-bit Linux machines. 
+
+These libraries are common to all Linux installations and should be installed
+prior to using Theano.
+
```

Comment:
1. I suggest we split them in two groups: the requirements that can be installed through pip (i.e. that the user can either install manually or let pip install for him while installing theano) and the others (python, python-dev, g++/clang, cuda, libgpuarray).

2. Move the section on how to install GPU here

3. Add "(optional)" on the side of the name of the optional requirements